pooch/downloaders.py

```diff
@@ -308,3 +314,113 @@ def callback(data):
             ftp.quit()
             if ispath:
                 output_file.close()
+
+
+class SFTPDownloader:  # pylint: disable=too-few-public-methods
+    """
+    Download manager for fetching files over SFTP.
+
+    When called, downloads the given file URL into the specified local file
+    Uses the :mod:`paramiko` module to manage downloads.
+
+    Use with :meth:`pooch.Pooch.fetch` to customize the download of files
+    (for example, to use authentication or print a progress bar).
+
+    Parameters
+    ----------
+    port : int
+        Port used for the SFTP connection.
+    username : str
+        User name used to login to the server. Only needed if the server
+        requires authentication (i.e., no anonymous SFTP).
+    password : str
+        Password used to login to the server. Only needed if the server
+        requires authentication (i.e., no anonymous SFTP). Use the empty
+        string to indicate no password is required.
+    timeout : int
+        Timeout in seconds for sftp socket operations, use None to mean no
+        timeout.
+    progressbar : bool
+        If True, will print a progress bar of the download to standard
+        error (stderr). Requires `tqdm <https://github.com/tqdm/tqdm>`__ to
+        be installed.
+
+    """
+
+    def __init__(
+        self,
+        port=22,
+        username="anonymous",
+        password="",
+        account="",
+        timeout=None,
+        progressbar=False,
+    ):
+        if paramiko is None:
+            raise ImportError("SFTP is only supported if paramiko is installed")
+
+        self.port = port
+        self.username = username
+        self.password = password
+        self.account = account
+        self.timeout = timeout
+        self.progressbar = progressbar
+        if self.progressbar and tqdm is None:
+            raise ValueError("Missing package 'tqdm' required for progress bars.")
```

Comment:
@luke-gregor the coverage report is saying that this line is never called in the tests.

Reply:
Added a new test for this - could not be "parametrized" with the existing test function as SFTPDownloader cannot be called if `paramiko` is not installed